### ISSUE TITLE
Warning and error while compiling driver

### ISSUE THREAD

**Issue by bk3a12** [arthur-liberman/linux_openvfd#11]
Hi,
I would like to ask I have encountered this warning and error while compiling driver.
warning: "CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND" is not defined, evaluates to 0 [-Wundef]
 #elif CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND
My tv box is tx3 mini running Armbian buster kernel version 5.9.0.
Could you please take a look and help me? Thank you!

**Comment by bk3a12** [comment]
This is the full log when I compiled.

sudo make ARCH=arm64 -C "/usr/src/linux-headers-5.9.0-arm-64" M="/home/dale/linux_openvfd/driver"              make: Entering directory '/usr/src/linux-headers-5.9.0-arm-64'
  AR      /home/dale/linux_openvfd/driver/built-in.a
  CC [M]  /home/dale/linux_openvfd/driver/protocols/i2c_sw.o
  CC [M]  /home/dale/linux_openvfd/driver/protocols/i2c_hw.o
  CC [M]  /home/dale/linux_openvfd/driver/protocols/spi_sw.o
  CC [M]  /home/dale/linux_openvfd/driver/controllers/dummy.o
  CC [M]  /home/dale/linux_openvfd/driver/controllers/seg7_ctrl.o
  CC [M]  /home/dale/linux_openvfd/driver/controllers/fd628.o
  CC [M]  /home/dale/linux_openvfd/driver/controllers/fd650.o
  CC [M]  /home/dale/linux_openvfd/driver/controllers/hd44780.o
  CC [M]  /home/dale/linux_openvfd/driver/controllers/gfx_mono_ctrl.o
  CC [M]  /home/dale/linux_openvfd/driver/controllers/ssd1306.o
  CC [M]  /home/dale/linux_openvfd/driver/controllers/pcd8544.o
  CC [M]  /home/dale/linux_openvfd/driver/controllers/il3829.o
  CC [M]  /home/dale/linux_openvfd/driver/openvfd_drv.o
/home/dale/linux_openvfd/driver/openvfd_drv.c:43:7: warning: "CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND" is not defined, evaluates to 0 [-Wundef]
 #elif CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  LD [M]  /home/dale/linux_openvfd/driver/openvfd.o
  MODPOST /home/dale/linux_openvfd/driver/Module.symvers
ERROR: modpost: "__stack_chk_guard" [/home/dale/linux_openvfd/driver/openvfd.ko] undefined!
make[1]: *** [scripts/Makefile.modpost:111: /home/dale/linux_openvfd/driver/Module.symvers] Error 1
make[1]: *** Deleting file '/home/dale/linux_openvfd/driver/Module.symvers'
make: *** [Makefile:1697: modules] Error 2
make: Leaving directory '/usr/src/linux-headers-5.9.0-arm-64'

**Comment by arthur-liberman** [comment]
`CONFIG_AMLOGIC_LEGACY_EARLY_SUSPEND` is specific to Amlogic linux kernel, and should not cause any problems, at least as far as I know.
Regarding the error you see, maybe this will shed a bit more light on the problem. I think it's due to your env configuration.
https://github.com/openzfs/zfs/issues/10985#issuecomment-705420322

**Comment by bk3a12** [comment]
HI, thanks for your advice. I've switched into focal and after that I could re compile the driver. But when I tried to load module with insmod, I got below error, which I didn't know where it came from.

`Message from syslogd@localhost at Jul 18 01:27:28 ...
 kernel:[   57.778976] Internal error: Oops: 96000004 [#1] PREEMPT SMP

Message from syslogd@localhost at Jul 18 01:27:28 ...
 kernel:[   58.028944] Code: 39016420 f9401a60 f9404fe1 f9400000 (f9406021)
Segmentation fault`
I could see it seemed to be loaded by lsmod but I have no idea whether it is okay. I tried OpenVFDService & and no luck.
And other question, when it comes to armbian, can I put vfd.conf file in the directory /storage/.config/ as we do for coreelec?
Thanks.
